### DISCUSSION TITLE
User More Faster Project Dependencies

### DISCUSSION THREAD

**Discussion by sirinath** [activej/activej#190] [Ideas]
I noticed the following dependencies 

- RocksDB
- Gson
- HPPC
- etc.

Also wondering if Logback can be changed to Log4J2 as this claimed to be more performant. Also, consider [ZeroLog](https://github.com/obsidiandynamics/zerolog).

IMDB (https://github.com/lmdbjava/lmdbjava) is noted to be more performant than RocksDB: https://github.com/lmdbjava/benchmarks/blob/master/results/20160710/README.md

Also might be worth considerin: https://github.com/castortech/mdbxjni

Also, https://github.com/ngs-doo/dsl-json seems to be more performant, than Gson, according to: https://github.com/fabienrenaud/java-json-benchmark. Also, noteworthy: https://github.com/RuedigerMoeller/fast-serialization. 

For binary serialisation: https://github.com/real-logic/simple-binary-encoding, https://chronoxor.github.io/FastBinaryEncoding/, http://cbor.io/, https://arrow.apache.org/. Binary benchmarks: https://speice.io/2019/09/binary-format-shootout.html 

Also following are some benchmarks on primitive collections: https://github.com/austinv11/Long-Map-Benchmarks, https://ivanursul.com/articles/java/collections-performance.html, http://java-performance.info/hashmap-overview-jdk-fastutil-goldman-sachs-hppc-koloboke-trove-january-2015/. It may be worthwhile to see if HPPC is the best to use.

**Comment by valerialistratova** [comment]
Hello, @sirinath
ActiveJ core maven components don’t have third-party dependencies. At the same time, more specific components require additional dependencies, some of them are optional.

For example, RocksDB is an optional dependency for creating memcache service, while ObjectWeb ASM is utilized only for ActiveSerializer and ActiveCodegen.
